### ISSUE TITLE
File::open on directory panics on Windows

### ISSUE THREAD

**Issue by russcam** [qdrant/wal#31]
The line

https://github.com/qdrant/wal/blob/0dd3943113ff7ec2fbc5428bb77ba206c8492fa9/src/lib.rs#L118

panics on Windows when `&path` is a directory; It seems that `File::open` does not work on directories on Windows. 

Here's a simple reproduction of the error exhibited in https://github.com/qdrant/qdrant/pull/922:

```rust
use std::fs::{create_dir_all, File};
use std::path::Path;
use fs2::FileExt;
use tempfile::Builder;

fn main() {
    let dir = Builder::new().prefix("foo").tempdir().unwrap();
    let sub_dir = Path::new(dir.path()).join("bar");
    create_dir_all(&sub_dir)
        .expect("Can't create sub directory");
    let dir = File::open(&sub_dir).unwrap(); // panics
    dir.try_lock_exclusive().unwrap();
}
```

**Comment by koopa1338** [comment]
Wanted to contribute with a PR, but this seems already been fixed in 765e9f91a5c5f0f1c9cf8f29d40d15b4f77f6d4d?

**Comment by generall** [comment]
I think this is already fixed in the latest versions, please feel free to re-open if not
